http/dotnet/httpclient/src/Middleware/RetryHandler.cs

```diff
@@ -0,0 +1,185 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Kiota.Http.HttpClient.Extensions;
+using Microsoft.Kiota.Http.HttpClient.Middleware.Options;
+
+namespace Microsoft.Kiota.Http.HttpClient.Middleware
+{
+    /// <summary>
+    /// A <see cref="DelegatingHandler"/> implementation using standard .NET libraries.
+    /// </summary>
+    public class RetryHandler : DelegatingHandler
```

Comment:
Respect retry-after: date form.

https://github.com/microsoft/kiota/issues/596

Reply:
Thanks for opening the issue.